File: app/src/main/java/com/github/goldy1992/mp3player/client/activities/FolderActivityInjector.java
package com.github.goldy1992.mp3player.client.activities;

import android.os.Bundle;

import com.github.goldy1992.mp3player.dagger.components.DaggerMediaActivityCompatComponent;
import com.github.goldy1992.mp3player.dagger.components.MediaActivityCompatComponent;

public class FolderActivityInjector extends FolderActivity {

    @Override
    public void onCreate(Bundle savedInstance) {
        initialiseDependencies();
        super.onCreate(savedInstance);
    }
    @Override
    void initialiseDependencies() {
        MediaActivityCompatComponent component = DaggerMediaActivityCompatComponent
                .factory()
                .create(getApplicationContext(), getWorkerId(), this);
        this.setMediaActivityCompatComponent(component);
        component.inject(this);

    }
}

File: app/src/main/java/com/github/goldy1992/mp3player/client/activities/FolderActivityInjector.kt
package com.github.goldy1992.mp3player.client.activities

import android.os.Bundle
import com.github.goldy1992.mp3player.commons.MikesMp3Player
import com.github.goldy1992.mp3player.client.dagger.components.DaggerMediaActivityCompatComponent

class FolderActivityInjector : FolderActivity() {
    public override fun onCreate(savedInstanceState: Bundle?) {
        initialiseDependencies()
        super.onCreate(savedInstanceState)
    }

    public override fun initialiseDependencies() {
        var app : MikesMp3Player = applicationContext!! as MikesMp3Player
        val component = DaggerMediaActivityCompatComponent
                .factory()
                .create(applicationContext,  this, app.getComponentClassMapper())
        mediaActivityCompatComponent = component
        component.inject(this)
    }
}